profiles/common/modules/custom/ecas/includes/ecas.inc

```diff
@@ -374,128 +375,144 @@ function ecas_sync_user_info(&$user, array $user_info, array $args) {
+    // If we don't have a valid user register them.
+    if (!$account && !empty($ecas_name)) {
 
-    // Also set an extra parameter for ecas_sync_user_info() -- this allows
-    // other modules to detect a user creation.
-    $ecas_sync_user_info_args['ecas_user_creation'] = TRUE;
-  }
+      $account = ecas_create_user($ecas_name);
 
-  // Final check to make sure we have a good user.
-  if ($account && $account->uid > 0) {
-    // Retrieve information about the user from ECAS attributes.
-    $user_info = phpCAS::getAttributes();
+      if ($account && $account->uid > 0) {
+        $_SESSION['ecas_goto'] = sprintf('user/%d/edit', $account->uid);
+      }
 
-    // Update her information.
-    if (!empty($user_info)) {
-      ecas_sync_user_info($account, $user_info, $ecas_sync_user_info_args);
-    }
-    else {
-      $args = array(':username' => $account->name);
-      drupal_set_message(t('Warning: no information could be found for username :username', $args));
-    }
+      // Set a session variable to denote this the initial login.
+      $_SESSION['ecas_first_login'] = TRUE;
 
-    // Case when account is blocked.
-    if ($account->status == 0) {
-      _ecas_session_end();
-      drupal_goto('ecas_warning_page');
+      // Also set an extra parameter for ecas_sync_user_info() -- this allows
+      // other modules to detect a user creation.
+      $ecas_sync_user_info_args['ecas_user_creation'] = TRUE;
     }
-    // When returning with same ECAS login as before closing Drupal session.
-    else {
-      if (
-        isset($account->data['ecas_login_date']) &&
-        $account->data['ecas_login_date'] == $user_info['cas:loginDate']
-      ) {
 
-        $msg = 'Rejected ECAS reused login: %n (ECAS)';
-        watchdog('user', $msg, array('%n' => $account->name), WATCHDOG_NOTICE);
+    // Final check to make sure we have a good user.
+    if ($account && $account->uid > 0) {
+      // Retrieve information about the user from ECAS attributes.
+      $user_info = phpCAS::getAttributes();
+
+      // Update her information.
+      if (!empty($user_info)) {
+        ecas_sync_user_info($account, $user_info, $ecas_sync_user_info_args);
+      }
+      else {
+        $args = array(':username' => $account->name);
+        drupal_set_message(t('Warning: no information could be found for username :username', $args));
+      }
 
-        _ecas_session_end();
+      if ($account->status == 0) {
+        // Case when account is blocked.
+        watchdog('user', 'Login denied for %ecas_name as blocked user.',
+          array('%ecas_name' => $ecas_name));
+        session_destroy();
 
-        $url = phpCAS::getServerLogoutURL() . '?url=' . _ecas_get_url_site_frontpage();
-        $msg = t('Your EU Login session needs to be renewed. For security reasons, please <a href="@url">logout and login again</a>.', ['@url' => $url]);
-        drupal_set_message($msg, 'error');
+        $user = drupal_anonymous_user();
```

Comment:
Why here we need to set $user = drupal_anonymous_user(); ?

In other places where login fails, we have only this methods:
watchdog()
session_destroy()
drupal_goto()

It should be the same everywhere, or explain why here it's different please.

Reply:
This has been removed